### PR TITLE
🔫 macos: set TCP_FASTOPEN_FORCE_ENABLE

### PR DESCRIPTION
**Pull request by database64128**
This disables the absolutely brutal TFO backoff mechanism, so that TFO can actually work.

### CODE REVIEW

File: src/listener.rs [hunk 1]
```diff
@@ -36,11 +39,25 @@ impl TfoListener {
         #[cfg(not(windows))]
         socket.set_reuseaddr(true)?;
 
+        // On all other platforms, TCP_FASTOPEN can be set before bind(), between bind() and listen(), or after listen().
+        // We prefer setting it before bind() as this feels most like the natural order of socket initialization sequence.
+        //
+        // On macOS, setting TCP_FASTOPEN_FORCE_ENABLE requires the socket to be in the TCPS_CLOSED state.
+        // TCP_FASTOPEN, on the other hand, can only be set when the socket is in the TCPS_LISTEN state.
+        cfg_if! {
+            if #[cfg(not(target_os = "macos"))] {
```
Comment (zonyitoo) [inline src/listener.rs]: I guess you may also want to enable it for iOS, watchOS, ...

File: src/listener.rs [hunk 2]
```diff
@@ -36,11 +39,25 @@ impl TfoListener {
...
+                set_tcp_fastopen(&socket)?;
+            } else {
+                set_tcp_fastopen_force_enable(&socket)?;
```
Comment (zonyitoo) [inline src/listener.rs]: If setsockopt `TCP_FASTOPEN_FORCE_ENABLE` fails, we shouldn't report an error here. Just use `if let Err(err)` to catch the error and print it in log

File: src/listener.rs (hunk 1 above)
Comment (database64128) [reply src/listener.rs]: Oops, I got too used to Go's `darwin` and forgot that in Rust we need to distinguish between these OSes. 😅

File: src/listener.rs (hunk 2 above)
Comment (database64128) [reply src/listener.rs]: Not sure what you mean. This isn't shadowsocks-rust. How are we supposed to log an error here and still return a listener? 😅

Comment (zonyitoo) [reply src/listener.rs]:
```rust
if let Err(err) = set_tcp_fastopen_force_enable(&socket) {
    debug!("JUST PRINT A LOG {:?}", err);
    // AND let it continue.
}
```

**Comment by zonyitoo** [comment]
Is this option could only applied to Listener?

File: src/listener.rs (hunk 2 above)
Comment (database64128) [reply src/listener.rs]: OK. I understand that the `log` crate is fairly "universal" and libraries are allowed to use it. However, is it really a good idea for low-level libraries like `tokio-tfo` to do debug logging like this? Does `tokio` also do this? What if the downstream user wants to fail on such occasions?

I'm very new to Rust, so I'll take whatever decision you make on this.

Comment (zonyitoo) [reply src/listener.rs]: My opinion is: `TCP_FASTOPEN_FORCE_ENABLE` is not the critical path for sockets to enable TCP Fast Open.

It is also Ok to just simply ignore the result:

```rust
let _ = set_tcp_fastopen_force_enable(&socket);
```

Comment (database64128) [reply src/listener.rs]: Added debug logging.

**Pull request merged**